### ISSUE TITLE
is.available() call in CBLHTTPServlet needs to be removed

### ISSUE THREAD

**Issue by yaronyg** [couchbase/couchbase-lite-java-listener#14]
# tldr

The use of the available() call doesn't honor the semantics of available and causes SSL support to fail. There is a fixme note next to the call that mentions a problem with GET not working if it's removed but I removed it and GET worked fine.
# Long Winded Version
## Semantics of Available()

By way of background let's very quickly review the intended semantics of Available(). The javadocs cover it pretty well but there is one line from them that is worth calling out: "In particular, an implementation that always returns 0 would be correct." 

So one can never, legally, run a check on Available() and expect a non-0 value. This point is actually reinforced in TJWS where Serve.ServerInputStream.available says:

``` java
    /**
     * Available bytes to read without blocking. If you are unlucky may
     * return 0 when there are more
     */
```
## Why I care

I am using SSL with the android listener and the SSL stream provider returned in Android (at least 4.3) always (as is explicitly legal) returns 0. This causes the Android Listener to think that PUT/POST requests with more than a few bytes of body are empty and to fail on parsing.
## The cause of the bug

Below is the code from CBLHTTPServlet that causes the SSL provider to fail:

``` java
        //fixme dont think i should have to call available here
        //but its blocking on get requests otherwise
        if(is != null && is.available() > 0) {
            conn.setDoInput(true);
            conn.setRequestInputStream(is);
        }
```

Because of how TJWS parses data if a request body is small enough it will actually work with this code. But if the request body gets too big (in the 100s of bytes) then the available() call will go through the TJWS code down to the raw stream which in the case of SSL on Android 4.3 always returns 0.
## Proposed fix

So I removed the if statement completely. After all if 'is' is null then we have bigger problems than not being able to get a request body. And available isn't correct. setDoInput does, literally, nothing near as I can tell looking at the code but setting it is harmless and looks right. And setRequestInputStream is harmless.

I ran tests both with and without SSL with GETs, PUTs, POSTs, DELETEs, HEADs, etc. Everything worked fine. I am testing against Android 4.3 (on x86 but that shouldn't matter). But I haven't been able to repo the GET issue mentioned in the fixme.
